### PR TITLE
Updates controller-runtime, remove WriteSVIDToDir

### PR DESCRIPTION
**Pull request by JoshVanL**
PR updates go modules, specifically controller-runtime to v0.16.3 so that the webhook server can be configured to use TLS options. This removes the need for the operator to write the SVID to disk and maintain it in memory.

PR removes the WriteSVIDToDir option from security.

Operator cache has been updated to use new controller-runtime types.

Related: #5756

### CODE REVIEW

File: tests/platforms/kubernetes/appmanager.go [hunk 1]
```diff
@@ -236,11 +236,12 @@ func (m *AppManager) WaitUntilJobState(isState func(*batchv1.Job, error) bool) (
 
 	var lastJob *batchv1.Job
 
-	waitErr := wait.PollImmediate(PollInterval, PollTimeout, func() (bool, error) {
```
Comment (JoshVanL) [inline tests/platforms/kubernetes/appmanager.go:239]: `wait.PollImmediate` has been deprecated.

File: pkg/operator/cache/cache.go [hunk 2]
```diff
@@ -36,81 +38,86 @@ var (
...
-				Label: podSelector,
-			},
+func GetFilteredCache(namespace string, podSelector labels.Selector) cache.NewCacheFunc {
+	return func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
+		// The only pods we are interested are in watchdog. we don't need to
+		// list/watch pods that we are almost sure have dapr sidecar already.
+		opts.ByObject = getTransformerFunctions(podSelector)
+		if len(namespace) > 0 {
+			opts.DefaultNamespaces = map[string]cache.Config{
+				namespace: {},
```
Comment (daixiang0) [inline pkg/operator/cache/cache.go:48]: Not use namespace, is it really useful?

Comment (JoshVanL) [reply pkg/operator/cache/cache.go:48]: Yes, this sets the namespaces which should be watched, [see here](https://github.com/kubernetes-sigs/controller-runtime/blob/2154ffbc22e26ffd8c3b713927f0df2fa40841f2/pkg/cache/cache.go#L193).

**Comment by yaron2** [comment]
> # Dapr Version Skew integration test (control-plane-master - 1.12.4)
> 🔗 **[Link to Action run](https://github.com/dapr/dapr/actions/runs/7576297148)**
> 
> Commit ref: [5621210](https://github.com/dapr/dapr/commit/5621210b9d90fb19098bdc7f1e0adc5db7fac09a)
> 
> ## ❌ Version Skew tests failed
> Please check the logs for details on the error.

@JoshVanL see this test failing

**Review by yaron2 — APPROVED** [review]

**Pull request merged**